### ISSUE TITLE
Scala 2.13 REPL has no color on Windows

### ISSUE THREAD

**Issue by Friendseeker** [VirtusLab/scala-cli#3394]
**Version(s)**
1.5.4

**Describe the bug**
Scala 2.13 REPL has no color on Windows

![image](https://github.com/user-attachments/assets/6a149344-452e-40a3-b3a8-deda13ac61c4)

**To Reproduce**
`scala-cli -S 2.13.15`

**Expected behaviour**
Output should be colored, like `sbt console`

![image](https://github.com/user-attachments/assets/81aa5e60-6bc9-429f-9092-329bee1a1e56)

c.c. @SethTisue

**Comment by Gedochao** [comment]
@SethTisue @lrytz @philwalk any clue where to look?

**Comment by SethTisue** [comment]
I wish I had some insight, but I don't.

**Comment by philwalk** [comment]
I verified that the problem exists in `cygwin64` running with `mintty`, and `msys64` running in `Windows Terminal`.
I'll  experiment with it to see if I can get any insights.

**Comment by philwalk** [comment]
This "feature" seems to be part of `scala 2.13` rather than `scala-cli` (same thing happens in any `scala 2.13` `REPL` session).

The following fix works for me when running either `bash` or `CMD` from `Windows Terminal`:
```
scala-cli -S 2.13 -Dscala.color=true
```
When directly running in a `CMD` session, the following works:
```
C:\opt\scala-2.13.15\bin>scala-cli -S 2.13 -Dscala.color=true -Djline.terminal=jline.WindowsTerminal
```

Not sure if these settings could easily be made to be the defaults in those environments.

<img width="537" alt="image" src="https://github.com/user-attachments/assets/04169b14-fe85-43cc-8d05-12e15913dbe6" />

This works in a `cygwin` mintty session:
![image](https://github.com/user-attachments/assets/e6c6bfb6-bc90-49d0-ae67-01dc16b66014)
 
and in a `Windows Terminal` msys64 bash session:
![image](https://github.com/user-attachments/assets/3b5c95ca-86b9-489a-b738-847f4a8c3eaa)

**Comment by SethTisue** [comment]
note that `scala.color` defaults to `auto`

but in `src/library/scala/util/Properties.scala` I see:

```
  private[scala] def coloredOutputEnabled: Boolean = propOrElse("scala.color", "auto") match {
    case "auto" => !isWin && consoleIsTerminal
    case s      => "" == s || "true".equalsIgnoreCase(s)
  }
```

note that the `!isWin`, which appears to be the culprit

the history for why this was added is at https://github.com/scala/scala/pull/5663 , which was back in 2017, in 2.12.x days

presumably a Scala 2.13 PR doing something better — not just in the Scala-CLI context — would have a good chance of being accepted

about the native-code angle, we've been on JLine 3 on 2.13.x for ages now, and https://github.com/scala/scala/pull/10898 aligns us with the JLine 3 project's current recommendations on Windows

I'm less confident that makes sense to consider changing 2.12, since with every year that passes we're less and less willing to touch 2.12 (which still uses JLine 2) — in general, but especially when it's something like the REPL that's fragile and requires manual testing

**Comment by Gedochao** [comment]
@SethTisue to confirm, is this something we should try to address in the Scala 2.13 (or less likely 2.12) REPL, rather than on the side of Scala CLI? Is there a ticket on Scala 2 side to track this, or should we raise one?

Adding `-Dscala.color=true` by default for REPL on Windows environments for Scala 2.x specifically in Scala CLI is something we could do, I guess... But maybe it's better to fix it at the source.

**Comment by SethTisue** [comment]
Agree. @Friendseeker @philwalk can I interest you in opening a ticket in scala/bug?

**Comment by philwalk** [comment]
Yes, the problem originates in the scala 2.13 REPL

**Comment by philwalk** [comment]
> @philwalk can I interest you in opening a ticket in scala/bug?

I'll look into it.   Not familiar with the build process for scala 2, but how hard can it be `:)`
If we change the default from "auto" to "true", people can do `-Dscala.color=false` if they run into problems.

**Comment by SethTisue** [comment]
> If we change the default from "auto" to "true", people can do -Dscala.color=false if they run into problems

I agree that a PR with only that change might well be accepted, assuming that you show the same manual testing that you showed above

**Comment by philwalk** [comment]
Here's the bug report: [https://github.com/scala/bug/issues/13077](https://github.com/scala/bug/issues/13077)

**Comment by philwalk** [comment]
I'm not sure why I seemed to need `-Djline.terminal=jline.WindowsTerminal` a few days ago, I can't reproduce it, so I assume it was confusion on my part.
The `scala 2.13` PR  https://github.com/scala/bug/issues/13077 seems to resolve the issue.

**Comment by Gedochao** [comment]
Closing this, nothing to do on Scala CLI side.
